capsules/src/bmp280.rs

```diff
@@ -0,0 +1,378 @@
+//! hil driver for Bmp280 Temperature and Pressure Sensor
+//!
+//! Written by Dorota <gihu.dcz@porcupinefactory.org>
+//!
+//! Based off the SHT3x code.
+//!
+//! Not implemented: pressure
+
+use core::cell::Cell;
+use kernel::debug;
+use kernel::hil;
+use kernel::hil::i2c;
+use kernel::hil::time::{Alarm, ConvertTicks};
+use kernel::utilities::cells::{OptionalCell, TakeCell};
+use kernel::ErrorCode;
+
+pub static BASE_ADDR: u8 = 0x76;
+
+/// Currently includes temperature only.
+/// This is the biggest buffer size in this module.
+pub const CALIBRATION_BYTES: usize = 6;
+
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+enum Register {
+    /// First register of calibration data.
+    /// Each register is 2 bytes long.
+    DIG_T1 = 0x88,
+    DIG_T2 = 0x8a,
+    DIG_T3 = 0x8c,
+    ID = 0xd0,
+    RESET = 0xe0,
+    /// measuring: [3]
+    /// im_update: [0]
+    STATUS = 0xf3,
+    /// osrs_t: [7:5]
+    /// osrs_p: [4:2]
+    /// mode: [1:0]
+    CTRL_MEAS = 0xf4,
+    /// t_sb: [7:5]
+    /// filter: [4:2]
+    /// spi3w_en: [0]
+    CONFIG = 0xf5,
+    PRESS_MSB = 0xf7,
+    PRESS_LSB = 0xf8,
+    /// xlsb: [7:4]
+    PRESS_XLSB = 0xf9,
+    TEMP_MSB = 0xfa,
+    TEMP_LSB = 0xfb,
+    /// xlsb: [7:4]
+    TEMP_XLSB = 0xfc,
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+struct CalibrationData {
+    dig_t1: u16,
+    dig_t2: i16,
+    dig_t3: i16,
+    // TODO: pressure calibration
+}
+
+/// CAUTION: calibration data puts least significant byte in the lowest address,
+/// readouts do the opposite.
+fn twobyte(lsb: u8, msb: u8) -> u16 {
```

Comment:
I disagree. For hinting the compiler to inline a given function out of performance considerations, the weak `#[inline]` attribute should be just fine. We should refrain from using `#[inline(always)]` and `#[inline(never)]` unless we have some empirical evidence this is necessary, because presumably the compiler has some good reason to disobey our recommendation.

Reply:
Actually, nevermind, I was confused by GitHub's weird suggestions and how they integrate into the review UI and didn't realize this was already addressed but not marked as `Outdated`.